src/driver.rs

```diff
@@ -410,6 +410,87 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::guess_drivers_for_write("test.gpkg", true)
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn guess_drivers_for_write(
+        filename: &str,
+        is_vector: bool,
+    ) -> impl Iterator<Item = Driver> + '_ {
+        let ext = {
+            let filename = filename.to_ascii_lowercase();
+            let e = match filename.rsplit_once('.') {
+                Some(("", _)) => "", // hidden file no ext
+                Some((f, "zip")) => {
+                    // zip files could be zipped shp or gpkg
+                    if f.ends_with(".shp") {
+                        "shp.zip"
+                    } else if f.ends_with(".gpkg") {
+                        "gpkg.zip"
+                    } else {
+                        "zip"
+                    }
+                }
+                Some((_, e)) => e, // normal file with ext
+                None => "",
+            };
+            e.to_string()
+        };
+
+        DriverManager::all()
+            .filter(move |d| {
+                let can_create = d.metadata_item("DCAP_CREATE", "").is_some()
+                    || d.metadata_item("DCAP_CREATECOPY", "").is_some();
+                let check_vector = is_vector && d.metadata_item("DCAP_VECTOR", "").is_some();
+                let check_raster = !is_vector && d.metadata_item("DCAP_RASTER", "").is_some();
+                let check_vector_translate =
+                    is_vector && d.metadata_item("DCAP_VECTOR_TRANSLATE_FROM", "").is_some();
+                (can_create && (check_vector || check_raster)) || (check_vector_translate)
+            })
+            .filter(move |d| {
+                if let Some(e) = &d.metadata_item("DMD_EXTENSION", "") {
+                    if *e == ext {
+                        return true;
+                    }
+                }
+                if let Some(e) = d.metadata_item("DMD_EXTENSIONS", "") {
+                    if e.split(' ').collect::<Vec<&str>>().contains(&ext.as_str()) {
```

Comment:
No need to `collect` here, there's `.find().is_some()` and `.any()`.

Reply:
Did `.any()`, `.find(),is_some()` got flagged by the clippy lol.